### PR TITLE
Jest testing

### PR DESCRIPTION
**Pull request by gchzhen**

### CODE REVIEW

File: client/index.jsx [hunk 1]
```diff
@@ -9,7 +9,7 @@ class Reviews extends React.Component {
     this.state = {
       reviews: []
     };
-    this.get = this.get.bind(this);
+    this.get = this.getReviews.bind(this);
```
Comment (matt-leyrat) [inline client/index.jsx]: This should probably be `this.getReviews` rather than `this.get`

File: client/setup-enzyme-adapter.js [hunk 2]
```diff
@@ -0,0 +1,4 @@
+import { configure } from 'enzyme';
+import Adapter from 'enzyme-adapter-react-16';
+
+configure({ adapter: new Adapter() });
```
Comment (matt-leyrat) [inline client/setup-enzyme-adapter.js]: Don't forget to add a newline at the end of all of your files! Do you have a eslint running in your IDE?

File: package.json [hunk 3]
```diff
@@ -11,7 +11,8 @@
     "build:dev": "webpack --watch",
     "start:dev": "nodemon server/index.js",
     "seed:db": "node database/seed.js",
-    "reset:db": "mysql -u root < database/schema.sql"
+    "reset:db": "mysql -u root < database/schema.sql",
+    "test": "jest --detectOpenHandles"
   },
   "dependencies": {
```
Comment (matt-leyrat) [inline package.json]: It looks like you've misplaced some of your dependencies into the `devDependencies` object (react, react-dom, etc.)

File: jest.config.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
+module.exports = {
+  verbose: true,
+  setupFilesAfterEnv: ['./client/setup-enzyme-adapter.js'],
+  transform: {
+    '^.+\\.jsx?$': 'babel-jest'
+  },
+  "moduleNameMapper": {
```
Comment (matt-leyrat) [inline jest.config.js]: Not sure if this is necessary, am I missing something?

File: public/bundle.js [hunk 5]
```diff
@@ -3,7 +3,7 @@
...
...ect.setPrototypeOf||function(e,t){return e.__proto__=t,e})(e,t)}var h=function(e){function t(e){var n,r,i;return function(e,t){if(!(e instanceof t))throw new TypeError("Cannot call a class as a function")}(this,t),r=this,(n=!(i=f(t).call(this,e))||"object"!==s(i)&&"function"!=typeof i?d(r):i).state={reviews:[]},n.get=n.get.bind(d(n)),n}var n,r,i;return function(e,t){if("function"!=typeof t&&null!==t)throw new TypeError("Super expression must either be null or a function");e.prototype=Object.create(t&&t.prototype,{constructor:{value:e,writable:!0,configurable:!0}}),t&&p(e,t)}(t,a.a.Component),n=t,(r=[{key:"componentDidMount",value:function(){var e=this;console.log("yes"),this.getReviews(function(t,n){t?window.alert("Error reviews not found",t):e.setState({reviews:n})})}},{key:"getReviews",value:function(e){var t=window.location.pathname.split("/")[2];u.a.ajax({url:"/api/listings/".concat(t,"/reviews"),success:function(t){e(null,t)},error:function(t){e(t)}})}},{key:"render",value:function(){return a.a.createElement("div",null,this.state.reviews)}}])&&c(n.prototype,r),i&&c(n,i),t}();i.a.render(a.a.createElement(h,null),document.getElementById("reviews"))},function(e,t,n){"use strict";
```
Comment (matt-leyrat) [inline public/bundle.js]: You shouldn't commit your bundle.js file.

File: public/index.html [hunk 6]
```diff
@@ -1,8 +1,5 @@
 <!doctype html>
-<html>
-    <meta http-equiv="Content-Security-Policy" content="default-src https://cdn.example.net; child-src 'none'; object-src 'none'">
-    <!-- <meta http-equiv="Content-Security-Policy" content="default-src 'self'; font-src 'self' data: fonts.gstatic.com; http://121.0.0:3004/"> -->
-    <!-- <meta http-equiv="Content-Security-Policy" content="font-src 'self' data:; img-src 'self' data:; default-src 'self' http://121.0.0:3004/"> -->
+<html></html>
```
Comment (matt-leyrat) [inline public/index.html]: lgtm

File: server/index.js [hunk 7]
```diff
@@ -32,5 +31,17 @@ app.get('/api/listings/:id/reviews', (req, res) => {
...
 
+app.get('/api/listings/', (req, res) => {
+  // let id = ParseInt(req.params.id);
+  getData((err, reviews) => {
+    if (err) {
+      res.status(404);
+      res.end();
+    } else {
+      res.status(200);
+      res.end(JSON.stringify(reviews));
```
Comment (matt-leyrat) [inline server/index.js]: Might want to look into using `res.send()` instead of `.end()` here.

File: specs/server.test.js [hunk 8]
```diff
@@ -0,0 +1,57 @@
+var express = require('express');
+var request = require('supertest');
+var $ = require('jquery');
+
+var app = express();
```
Comment (matt-leyrat) [inline specs/server.test.js]: I think these tests cover more than enough of the back-end, but we should shift to the Front End and test as we build.

File: webpack.config.js [hunk 9]
```diff
@@ -13,14 +13,25 @@ module.exports = {
...
           options: {
             presets: ['@babel/preset-env', '@babel/preset-react']
           }
         },
+      },
+      {
+        // Preprocess our own .css files
+        // This is the place to add your own loaders (e.g. sass/less etc.)
+        // for a list of loaders, see https://webpack.js.org/loaders/#styling
+        test: /\.css$/,
```
Comment (matt-leyrat) [inline webpack.config.js]: Are you planning on keeping your css in a styles.css file or using JSS / styled-components / etc. ?

File: babel.config.js [hunk 10]
```diff
@@ -3,5 +3,10 @@ module.exports = {
     '@babel/preset-env',
     '@babel/preset-react'
   ],
-  "plugins": ["babel-plugin-styled-components"]
+  "plugins": [
+    "babel-plugin-styled-components",
+    {
+      "ssr": false
```
[the remark this answers is not in the document]
Comment (gchzhen) [reply babel.config.js]: I've added it after you sent that link regarding styled components and server-side rendering today @matt-leyrat !

File: client/components/Review.jsx [hunk 11]
```diff
@@ -2,21 +2,46 @@ import React from 'react';
...
-      <div>LOCATION: {props.nameOfProps.location}</div>
-      <div>ATHLETIC TYPE: {props.nameOfProps.athletic_type}</div>
-      <div>AGE: {props.nameOfProps.age}</div>
-      <div>BODY TYPE: {props.nameOfProps.body_type}</div>
-      <div>WHAT YOU LIKE </div>
-      <div>WHAT YOU DIDN'T LIKE </div>
-      <div>FIT: {props.nameOfProps.fit}</div>
-      <div>{props.nameOfProps.date}</div>
-      <div>{props.nameOfProps.review_title}</div>
-      <div>{props.nameOfProps.review_details}</div>
```
[the remark this answers is not in the document]
Comment (gchzhen) [reply client/components/Review.jsx]: Thank you @matt-leyrat, will look into it today :)

File: client/components/Reviews.jsx [hunk 12]
```diff
@@ -42,27 +47,61 @@ class Reviews extends React.Component {
...
+    } else if (avg <= 4 && avg > 3) {
+        str = '<div id="average_star_4"></div>';
+    } else if (avg <= 3 && avg > 2) {
+      str = '<div id="average_star_3"></div>';
+    } else if (avg <= 2 && avg > 1) {
+      str = '<div id="average_star_2"></div>';
+    } else {
+      str = '<div id="average_star_1"></div>';
+    }
+    return <div>{ReactHtmlParser(str)}</div>;
```
[the remark this answers is not in the document]
Comment (gchzhen) [reply client/components/Reviews.jsx]: @matt-leyrat, yes, this is what I realized after talking with you! I thought I'm using modular and reusable CSS module! Looking forward to play around with styled components!

**Pull request merged**